xarray/plot/plot.py

```diff
@@ -449,10 +449,28 @@ def newplotfunc(darray, x=None, y=None, figsize=None, size=None,
         if imshow_rgb:
             # Don't add a colorbar when showing an image with explicit colors
             add_colorbar = False
+            # Calculate vmin and vmax automatically for `robust=True`
+            if robust:
+                if vmin is None:
+                    vmin = np.nanpercentile(darray, ROBUST_PERCENTILE)
+                if vmax is None:
+                    vmax = np.nanpercentile(darray, 100 - ROBUST_PERCENTILE)
+                robust = False
+            # Scale interval [vmin .. vmax] to [0 .. 1] and clip to bounds
+            if vmin is not None or vmax is not None:
+                vmin = vmin if vmin is not None else darray.min()
+                vmax = vmax if vmax is not None else darray.max()
+                darray = ((darray.astype('f8') - vmin) / (vmax - vmin))
+                vmin, vmax = None, None
+                # There's a cyclic dependency via DataArray, so we can't
+                # import xarray.ufuncs in global or outer scope.
+                import xarray.ufuncs as xu
+                darray = xu.minimum(xu.maximum(darray.astype('f4'), 0), 1)
```

Comment:
do we need to type cast this back to a 4byte float?

Reply:
@shoyer - do you have a solution off had for the cyclic import problem here?

Reply:
We don't *need* to downcast, in that it will work even if we don't, but downcasting ~halves memory use further inside matplotlib which can be a 5-10x multiple of the input data size.  Happy to remove if you don't think that's worth it.

The best alternative solution I can see is refactoring `xarray.ufuncs` to do all the imports inside a function or class, and pay the performance cost when creating ufuncs and slightly increased overhead in dispatch.  Personally, I prefer a slight slowdown when creating scaled RGB images - much less common, and it's probably negligible given the array operations involved anyway.

Reply:
I'm happy with downcasting, let's just add a comment to explain why.

I don't think we can avoid the cycle import issue. This is a downside of our method heavy design. However, it would be nice if we moved the import statement to the top of the method.